### PR TITLE
Closes #1275: Vitamins, minerals, amino acid and others tags support added

### LINKED ISSUE #1275: Add support for vitamins and minerals in products
We recently added support for vitamins and minerals in products.
In some cases, you will find what is below in a product:
```
Added vitamins:
Minerals:
Added amino acids:
Other nutritional substances added:
```

===== Vitamins and minerals detection =====

====== Vitamins ======
<pre>"vitamins_tags":["en:vitamin-a","en:cholecalciferol","en:vitamin-e","en:vitamin-c","en:thiamin","en:riboflavin","en:niacin","en:vitamin-b6","en:folic-acid","en:vitamin-b12","en:biotin","en:phylloquinone"]</pre>
====== Minerals ======
<pre>"minerals_tags":["en:calcium-citrate","en:potassium-citrate","en:sodium-citrate","en:magnesium-chloride","en:calcium-phosphate","en:potassium-phosphate","en:ferrous-sulphate","en:zinc-sulphate","en:cupric-sulphate","en:manganese-sulphate","en:sodium-carbonate","en:potassium-iodide"]</pre>

====== Amino acids ======
<pre>"amino_acids_tags":["en:l-tryptophan"]</pre>

====== Other nutritional substances =====
<pre>"other_nutritional_substances_tags":["en:taurine","en:inositol","en:choline-citrate","en:taurine","en:inositol","en:choline-citrate","en:taurine","en:inositol","en:choline-citrate"]<pre>

### PR DESCRIPTION
**Pull request by amanjeetsingh150**
## Description

Added the support for the vitamin, mineral, amino Acid and other substances tags in ingredient tab of ProductActivity.

## Related issues and discussion
#1275 
 
 ## Screen-shots, if any
 
<br>
 
<br>
 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [x] Code is well documented
 - [x] Include unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [x] Code passes Travis builds in your branch
 - [x] If you have multiple commits please combine them into one commit by squashing them.
 - [x] Read and understood the contribution guidelines .

### CODE REVIEW

**Comment by amanjeetsingh150** [comment]
@teolemon @Karljoones please review :)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java [hunk 1]
```diff
@@ -118,6 +127,66 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
...
+        List<String> mineralTags = product.getMineralTags();
+        List<String> otherNutritionTags = product.getOtherNutritionTags();
+        String prefix = " ";
+
+        if (!vitaminTagsList.isEmpty()) {
+            StringBuilder vitaminStringBuilder = new StringBuilder();
+            vitaminTagsTextView.append(bold(getString(R.string.vitamin_tags_text)));
+            for (String vitamins : vitaminTagsList) {
+                vitaminStringBuilder.append(prefix);
+                prefix = ",";
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: One more tiny thing, can you add a space after any commas that you use?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java [hunk 2]
```diff
@@ -118,6 +127,66 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
...
+            vitaminTagsTextView.setVisibility(View.GONE);
+        }
+
+        if (!aminoAcidTagsList.isEmpty()) {
+            String aminoPrefix = " ";
+            StringBuilder aminoAcidStringBuilder = new StringBuilder();
+            aminoAcidTagsTextView.append(bold(getString(R.string.amino_acid_tags_text)));
+            for (String aminoAcid : aminoAcidTagsList) {
+                aminoAcidStringBuilder.append(aminoPrefix);
+                aminoPrefix = ",";
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: Same here

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java [hunk 3]
```diff
@@ -118,6 +127,66 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
...
+            aminoAcidTagsTextView.setVisibility(View.GONE);
+        }
+
+        if (!mineralTags.isEmpty()) {
+            String mineralPrefix = " ";
+            StringBuilder mineralsStringBuilder = new StringBuilder();
+            mineralTagsTextView.append(bold(getString(R.string.mineral_tags_text)));
+            for (String mineral : mineralTags) {
+                mineralsStringBuilder.append(mineralPrefix);
+                mineralPrefix = ",";
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: And here

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java [hunk 4]
```diff
@@ -118,6 +127,66 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
...
+            mineralTagsTextView.setVisibility(View.GONE);
+        }
+
+        if (!otherNutritionTags.isEmpty()) {
+            String otherNutritionPrefix = " ";
+            StringBuilder otherNutritionStringBuilder = new StringBuilder();
+            otherNutritionTagTextView.append(bold(getString(R.string.other_tags_text)));
+            for (String otherSubstance : otherNutritionTags) {
+                otherNutritionStringBuilder.append(otherNutritionPrefix);
+                otherNutritionPrefix = ",";
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: And here

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java [hunk 5]
```diff
@@ -231,6 +300,10 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         }
     }
 
+    private String trimLanguagePartFromString(String string) {
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: Can you document this? I know it's a small method but I would like you to document what you use it for

Comment (amanjeetsingh150) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: On it :+1:

Comment (amanjeetsingh150) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java]: Update my PR here are the resultant Screenshots:
<br>

**Comment by Karljoones** [comment]
Fixed the conflicts

**Comment by amanjeetsingh150** [comment]
@Karljoones Are there still any conflicts?

**Comment by Karljoones** [comment]
Nope, I resolved them

**Comment by amanjeetsingh150** [comment]
@Karljoones Hey the Travis build is failing. I gave a look at the logs. But couldn't find something unusual, Can you help me to figure out this?

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -487,7 +487,14 @@
     <string name="connection">Connected</string>
     <string name="contributor_string">Contributor</string>
 
+    <!--Vitamin, Amino Acid, mineral and other nutrition Tags TextView-->
+    <string name="vitamin_tags_text">Vitamin Categories:</string>
+    <string name="amino_acid_tags_text">Amino Acid Categories:</string>
+    <string name="mineral_tags_text">Minerals Category:</string>
+    <string name="other_tags_text">Other Substance:</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Other substances:

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -487,7 +487,14 @@
     <string name="connection">Connected</string>
     <string name="contributor_string">Contributor</string>
 
+    <!--Vitamin, Amino Acid, mineral and other nutrition Tags TextView-->
+    <string name="vitamin_tags_text">Vitamin Categories:</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Vitamins:

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -487,7 +487,14 @@
...
+    <string name="amino_acid_tags_text">Amino Acid Categories:</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Amino acids:

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -487,7 +487,14 @@
...
+    <string name="mineral_tags_text">Minerals Category:</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Minerals:

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -487,7 +487,14 @@
...
+    <string name="other_tags_text">Other Substance:</string>
+
     <!---->
     <string name="product_transition" translatable="false">productTransition</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Not your fault but: productTransition shouldn't be here

Comment (amanjeetsingh150) [reply app/src/main/res/values/strings.xml]: @teolemon should I shift it above?

**Comment by amanjeetsingh150** [comment]
@teolemon Thank you for the review I have updated the strings accordingly. Please review.
Thanks :+1:

**Comment by Karljoones** [comment]
@amanjeetsingh150 Any chance of a final screenshot before merge?

**Comment by amanjeetsingh150** [comment]
@Karljoones Sure sir. These are two examples:

<br>

**Comment by Karljoones** [comment]
Merged, thanks @amanjeetsingh150

**Pull request merged**